### PR TITLE
Setup Hardhat, Prettier & Lint

### LINKED ISSUE #69: Setup Hardhat Project
### Overview

Since we're looking at putting funds for Gigs into a smart contract, then we need to setup the Hardhat project into the repo. It will be using TypeScript and all them goodies, i.e. it will use the Advanced TypeScript setup from Hardhat template.

### PR DESCRIPTION
**Pull request by carlomigueldy**
Not much information to say about this pull request but it's just setting up the Hardhat Project using the Advanced TypeScript template that Hardhat provides and I've added some scripts as well.

- My code changes/additions are mostly the scripts and bumped up dependency versions.
- For the most code, smart contract, tests, deploy scripts are all generated from the Hardhat template.

Closes #69

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,14 +5,28 @@
...
+    "deploy:local": "yarn workspace hardhat-project deploy:local",
+    "deploy:rinkeby": "yarn workspace hardhat-project deploy:rinkeby",
+    "deploy:mainnet": "yarn workspace hardhat-project deploy:mainnet",
+    "compile": "yarn workspace hardhat-project compile",
+    "prettier": "yarn workspace hardhat-project prettier",
+    "accounts": "yarn workspace hardhat-project accounts",
+    "clean": "yarn workspace hardhat-project clean",
+    "coverage": "yarn workspace hardhat-project coverage",
+    "eslint": "yarn workspace hardhat-project eslint",
+    "solhint": "yarn workspace hardhat-project solhint"
```
Comment (frankTurtle) [inline package.json:23]: "Some scripts" 😆

**Review by frankTurtle — APPROVED** [review]
🤙🤙

Schweet :) Learned some basic unit testing from this!

File: package.json (hunk 1 above)
Comment (carlomigueldy) [reply package.json:23]: Yeah 😆

**Comment by carlomigueldy** [comment]
> 🤙🤙
> 
> Schweet :) Learned some basic unit testing from this!

Sounds good! That comes along with the Hardhat template, but I would definitely be writing a bunch of unit tests in this smart contract, probably some mocks if need may be

**Pull request merged**